Compatibility with Kirkstone

Hi @OldManYellsAtCloud - I noticed recent versions only have compatibility with mickledore

LAYERSERIES_COMPAT_firefox-browser-layer = "mickledore"

Is there a reason for this, or can I try to build them with kirkstone? 
If so, how can I build a recent version with Kirkstone?

@OldManYellsAtCloud  understood ... I just tried against kirkstone and failed. It's a lot of output, but I will try to include the most important lines:

`
ERROR: nss-3.89-3.89-r0 do_compile: ExecutionError('/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/temp/run.do_compile.181965', 2, None, None)
ERROR: Logfile of failure stored in: /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/temp/log.do_compile.181965
Log data follows:
| DEBUG: Executing shell function do_compile
| make: Entering directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss'
| coreconf/config.mk:138: CPU_ARCH is not x86_64, disabling -mavx2
| make all
| make[1]: Entering directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss'
| coreconf/config.mk:138: CPU_ARCH is not x86_64, disabling -mavx2
| # no real way to encode these in any sensible way
| make -C coreconf/nsinstall program
| make[2]: Entering directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss/coreconf/nsinstall'
| ../../coreconf/config.mk:138: CPU_ARCH is not x86_64, disabling -mavx2
| gcc  -o Linux3.4_arm_glibc_PTH_OPT.OBJ/nsinstall.o -c -std=c99 -O2 -isystem/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native/usr/include -O2 -pipe -DLINUX -Dlinux -DGETCWD_CANT_MALLOC -Wall -Wshadow -DNSS_NO_GCC48 -DXP_UNIX -DXP_UNIX -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -DNSS_DISABLE_AVX2 -DNSS_NO_INIT_SUPPORT -DUSE_UTIL_DIRECTLY -DNO_NSPR_10_SUPPORT -DSSL_DISABLE_DEPRECATED_CIPHER_SUITE_NAMES -I/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot/usr/include/nspr -I../../../dist/Linux3.4_arm_glibc_PTH_OPT.OBJ/include -I../../../dist/public/coreconf -I../../../dist/private/coreconf  nsinstall.c
| gcc  -o Linux3.4_arm_glibc_PTH_OPT.OBJ/pathsub.o -c -std=c99 -O2 -isystem/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native/usr/include -O2 -pipe -DLINUX -Dlinux -DGETCWD_CANT_MALLOC -Wall -Wshadow -DNSS_NO_GCC48 -DXP_UNIX -DXP_UNIX -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -DNSS_DISABLE_AVX2 -DNSS_NO_INIT_SUPPORT -DUSE_UTIL_DIRECTLY -DNO_NSPR_10_SUPPORT -DSSL_DISABLE_DEPRECATED_CIPHER_SUITE_NAMES -I/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot/usr/include/nspr -I../../../dist/Linux3.4_arm_glibc_PTH_OPT.OBJ/include -I../../../dist/public/coreconf -I../../../dist/private/coreconf  pathsub.c
| rm -f Linux3.4_arm_glibc_PTH_OPT.OBJ/nsinstall
| gcc  -o Linux3.4_arm_glibc_PTH_OPT.OBJ/nsinstall -O2 -isystem/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native/usr/include -O2 -pipe -DLINUX -Dlinux -DGETCWD_CANT_MALLOC -Wall -Wshadow -DNSS_NO_GCC48 -DXP_UNIX -DXP_UNIX -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -DNSS_DISABLE_AVX2 -DNSS_NO_INIT_SUPPORT -DUSE_UTIL_DIRECTLY -DNO_NSPR_10_SUPPORT -DSSL_DISABLE_DEPRECATED_CIPHER_SUITE_NAMES -I/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot/usr/include/nspr -I../../../dist/Linux3.4_arm_glibc_PTH_OPT.OBJ/include -I../../../dist/public/coreconf -I../../../dist/private/coreconf  Linux3.4_arm_glibc_PTH_OPT.OBJ/nsinstall.o Linux3.4_arm_glibc_PTH_OPT.OBJ/pathsub.o    -lpthread  -ldl -lc
| true -m 775 Linux3.4_arm_glibc_PTH_OPT.OBJ/nsinstall ../../../dist/Linux3.4_arm_glibc_PTH_OPT.OBJ/bin
| make[2]: Leaving directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss/coreconf/nsinstall'
| make export
| make[2]: Entering directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss'
| coreconf/config.mk:138: CPU_ARCH is not x86_64, disabling -mavx2
| make[3]: Entering directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss/coreconf'
| ../coreconf/config.mk:138: CPU_ARCH is not x86_64, disabling -mavx2
| make[4]: Entering directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss/coreconf/nsinstall'
| ../../coreconf/config.mk:138: CPU_ARCH is not x86_64, disabling -mavx2
| make[4]: Nothing to be done for 'export'.
| make[4]: Leaving directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss/coreconf/nsinstall'
| make[3]: Leaving directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss/coreconf'
| make[3]: Entering directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss/lib'
| ../coreconf/config.mk:138: CPU_ARCH is not x86_64, disabling -mavx2
| make[4]: Entering directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss/lib/base'
| ../../coreconf/config.mk:138: CPU_ARCH is not x86_64, disabling -mavx2
| ../../coreconf/nsinstall/Linux3.4_arm_glibc_PTH_OPT.OBJ/nsinstall -R -m 444 baset.h ../../../dist/private/nss
| ../../coreconf/nsinstall/Linux3.4_arm_glibc_PTH_OPT.OBJ/nsinstall -R -m 444 base.h ../../../dist/private/nss
| ../../coreconf/nsinstall/Linux3.4_arm_glibc_PTH_OPT.OBJ/nsinstall -R -m 444 nssbaset.h ../../../dist/public/nss
| ../../coreconf/nsinstall/Linux3.4_arm_glibc_PTH_OPT.OBJ/nsinstall -R -m 444 nssbase.h ../../../dist/public/nss

...


| make[4]: Entering directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss/lib/base'
| ../../coreconf/config.mk:138: CPU_ARCH is not x86_64, disabling -mavx2
| arm-poky-linux-gnueabi-gcc  -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a7 -Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -fmacro-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native=  -Wl,-z,relro,-z,now -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security --sysroot=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native=  -o Linux3.4_arm_glibc_PTH_OPT.OBJ/arena.o -c -std=c99 -O2 -fPIC   -pipe -ffunction-sections -fdata-sections -DHAVE_STRERROR -DLINUX -Dlinux -Wall -Wshadow -DNSS_NO_GCC48 -DXP_UNIX -DXP_UNIX -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -DNSS_DISABLE_AVX2 -DNSS_NO_INIT_SUPPORT -DUSE_UTIL_DIRECTLY -DNO_NSPR_10_SUPPORT -DSSL_DISABLE_DEPRECATED_CIPHER_SUITE_NAMES -I/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot/usr/include/nspr -I../../../dist/Linux3.4_arm_glibc_PTH_OPT.OBJ/include -I../../../dist/public/nss -I../../../dist/private/nss -I../../../dist/public/nspr  arena.c
| arm-poky-linux-gnueabi-gcc  -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a7 -Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -fmacro-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native=  -Wl,-z,relro,-z,now -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security --sysroot=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native=  -o Linux3.4_arm_glibc_PTH_OPT.OBJ/error.o -c -std=c99 -O2 -fPIC   -pipe -ffunction-sections -fdata-sections -DHAVE_STRERROR -DLINUX -Dlinux -Wall -Wshadow -DNSS_NO_GCC48 -DXP_UNIX -DXP_UNIX -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -DNSS_DISABLE_AVX2 -DNSS_NO_INIT_SUPPORT -DUSE_UTIL_DIRECTLY -DNO_NSPR_10_SUPPORT -DSSL_DISABLE_DEPRECATED_CIPHER_SUITE_NAMES -I/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot/usr/include/nspr -I../../../dist/Linux3.4_arm_glibc_PTH_OPT.OBJ/include -I../../../dist/public/nss -I../../../dist/private/nss -I../../../dist/public/nspr  error.c
| arm-poky-linux-gnueabi-gcc  -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a7 -Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -fmacro-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native=  -Wl,-z,relro,-z,now -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -W

...

error=format-security --sysroot=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native=  -o Linux3.4_arm_glibc_PTH_OPT.OBJ/selfserv.o -c -std=c99 -O2 -fPIC   -pipe -ffunction-sections -fdata-sections -DHAVE_STRERROR -DLINUX -Dlinux -Wall -Wshadow -DNSS_NO_GCC48 -DXP_UNIX -DXP_UNIX -DNSPR20 -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -DNSS_DISABLE_AVX2 -DNSS_NO_INIT_SUPPORT -DUSE_UTIL_DIRECTLY -DNO_NSPR_10_SUPPORT -DSSL_DISABLE_DEPRECATED_CIPHER_SUITE_NAMES -I/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot/usr/include/nspr -I../../../dist/Linux3.4_arm_glibc_PTH_OPT.OBJ/include -I../../../dist/public/nss -I../../../dist/private/nss -I../../../dist/public/seccmd -I../../../dist/public/dbm  selfserv.c
| selfserv.c: In function 'getBoundListenSocket':
| selfserv.c:1724:16: warning: implicit declaration of function 'PR_GetPrefLoopbackAddrInfo' [-Wimplicit-function-declaration]
|  1724 |     prStatus = PR_GetPrefLoopbackAddrInfo(&addr, port);
|       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
| rm -f Linux3.4_arm_glibc_PTH_OPT.OBJ/selfserv
| arm-poky-linux-gnueabi-gcc  -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a7 -Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -fmacro-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native=  -Wl,-z,relro,-z,now -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security --sysroot=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native=  -o Linux3.4_arm_glibc_PTH_OPT.OBJ/selfserv -O2 -fPIC   -pipe -ffunction-sections -fdata-sections -DHAVE_STRERROR -DLINUX -Dlinux -Wall -Wshadow -DNSS_NO_GCC48 -DXP_UNIX -DXP_UNIX -DNSPR20 -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -UDEBUG -DNDEBUG -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE -DSDB_MEASURE_USE_TEMP_DIR -D_REENTRANT -DNSS_DISABLE_AVX2 -DNSS_NO_INIT_SUPPORT -DUSE_UTIL_DIRECTLY -DNO_NSPR_10_SUPPORT -DSSL_DISABLE_DEPRECATED_CIPHER_SUITE_NAMES -I/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot/usr/include/nspr -I../../../dist/Linux3.4_arm_glibc_PTH_OPT.OBJ/include -I../../../dist/public/nss -I../../../dist/private/nss -I../../../dist/public/seccmd -I../../../dist/public/dbm  Linux3.4_arm_glibc_PTH_OPT.OBJ/selfserv.o -Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -fmacro-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0=/usr/src/debug/nss-3.89/3.89-r0                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native=  -Wl,-z,relro,-z,now  -z noexecstack  -z noexecstack  -z noexecstack  -z noexecstack  -z noexecstack  -z noexecstack  -z noexecstack  -z noexecstack  -z noexecstack  -z noexecstack ../../../dist/Linux3.4_arm_glibc_PTH_OPT.OBJ/lib/libsectool.a  -L../../../dist/Linux3.4_arm_glibc_PTH_OPT.OBJ/lib -L../../../dist/Linux3.4_arm_glibc_PTH_OPT.OBJ/lib -lnssutil3 -L../../../dist/Linux3.4_arm_glibc_PTH_OPT.OBJ/lib -lplc4 -lplds4 -lnspr4  -lssl3 -lsmime3 -lnss3 -lpthread  -ldl -lc
| /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/recipe-sysroot-native/usr/bin/arm-poky-linux-gnueabi/../../libexec/arm-poky-linux-gnueabi/gcc/arm-poky-linux-gnueabi/11.3.0/ld: Linux3.4_arm_glibc_PTH_OPT.OBJ/selfserv.o: in function `getBoundListenSocket':
| /usr/src/debug/nss-3.89/3.89-r0/nss-3.89/nss/cmd/selfserv/selfserv.c:1724: undefined reference to `PR_GetPrefLoopbackAddrInfo'
| collect2: error: ld returned 1 exit status
| make[4]: Leaving directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss/cmd/selfserv'
| make[4]: *** [../../coreconf/rules.mk:165: Linux3.4_arm_glibc_PTH_OPT.OBJ/selfserv] Error 1
| make[3]: Leaving directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss/cmd'
| make[3]: *** [../coreconf/rules.mk:44: selfserv] Error 2
| make[2]: Leaving directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss'
| make[2]: *** [coreconf/rules.mk:44: cmd] Error 2
| make[1]: *** [manifest.mn:21: all] Error 2
| make[1]: Leaving directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss'
| make: *** [coreconf/rules.mk:24: autobuild] Error 2
| make: Leaving directory '/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/nss-3.89/3.89-r0/nss-3.89/nss'
| WARNING: exit code 2 from a shell command.

`

@OldManYellsAtCloud - THANK YOU! I tried switching my build to mickledore in the meantime b/c it was getting to be quite a lot of changes for kirkstone to build. However, now that I am on mickledore, I tried to build and got an error!!!

Here's my config:
```
  # main branch - commit 2023-09-04
  meta-browser:
    url: "https://github.com/OldManYellsAtCloud/meta-browser.git"
    commit: e4cd3f4f3942bc2ec16f0b9fb68078d34bc4b5c4
    layers:
      meta-firefox:
  
  # mickledore branch - commit 2023-06-22
  meta-clang:
    url: "https://github.com/kraj/meta-clang.git"
    commit: af4dcba009ba98250315520f3003fde4ee164cce

    # firefox
    PACKAGECONFIG:append:firefox = " webgl gpu wayland alsa"
```
all my other layers are compatible with mickledore, or are on a mickledore branch

Here's the error I just got: 
```
ERROR: firefox-117.0-r0 do_configure: ExecutionError('/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/temp/run.do_configure.1442186', 1, None, None)
ERROR: Logfile of failure stored in: /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/temp/log.do_configure.1442186
Log data follows:
| DEBUG: Executing python function extend_recipe_sysroot
| NOTE: Direct dependencies are ['/home/chris/yocto-3/build/../meta-browser/meta-firefox/recipes-devtools/nss/nss-3.89.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../meta-browser/meta-firefox/recipes-devtools/python-zstandard/python-zstandard_0.21.0.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../meta-browser/meta-firefox/recipes-devtools/wasi-sdk/wasi-sdk-native_git.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../meta-clang/recipes-devtools/clang/clang-cross_git.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../meta-openembedded/meta-oe/recipes-gnome/gtk+/gtk+_2.24.33.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-core/dbus/dbus-glib_0.112.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-core/glibc/glibc_2.37.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-devtools/gcc/gcc-runtime_12.3.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-devtools/gcc/libgcc_12.3.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-devtools/quilt/quilt-native_0.67.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-devtools/rust/libstd-rs_1.68.2.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-gnome/gtk+/gtk+3_3.24.36.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-graphics/mesa/mesa_23.0.3.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-graphics/xorg-lib/libxi_1.8.1.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-graphics/xorg-lib/libxt_1.2.1.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-multimedia/alsa/alsa-lib_1.2.8.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-multimedia/pulseaudio/pulseaudio_16.1.bb:do_populate_sysroot', '/home/chris/yocto-3/build/../poky/meta/recipes-support/libevent/libevent_2.1.12.bb:do_populate_sysroot', 'virtual:native:/home/chris/yocto-3/build/../meta-browser/meta-firefox/recipes-devtools/cbindgen/cbindgen_0.24.3.bb:do_populate_sysroot', 'virtual:native:/home/chris/yocto-3/build/../meta-clang/recipes-devtools/clang/clang_git.bb:do_populate_sysroot', 'virtual:native:/home/chris/yocto-3/build/../meta-openembedded/meta-oe/recipes-devtools/nodejs/nodejs_18.17.1.bb:do_populate_sysroot', 'virtual:native:/home/chris/yocto-3/build/../poky/meta/recipes-devtools/gnu-config/gnu-config_git.bb:do_populate_sysroot', 'virtual:native:/home/chris/yocto-3/build/../poky/meta/recipes-devtools/patch/patch_2.7.6.bb:do_populate_sysroot', 'virtual:native:/home/chris/yocto-3/build/../poky/meta/recipes-devtools/pkgconfig/pkgconfig_git.bb:do_populate_sysroot', 'virtual:native:/home/chris/yocto-3/build/../poky/meta/recipes-devtools/pseudo/pseudo_git.bb:do_populate_sysroot', 'virtual:native:/home/chris/yocto-3/build/../poky/meta/recipes-devtools/python/python3_3.11.2.bb:do_populate_sysroot', 'virtual:native:/home/chris/yocto-3/build/../poky/meta/recipes-devtools/rust/cargo_1.68.2.bb:do_populate_sysroot', 'virtual:native:/home/chris/yocto-3/build/../poky/meta/recipes-devtools/rust/rust_1.68.2.bb:do_populate_sysroot', 'virtual:native:/home/chris/yocto-3/build/../poky/meta/recipes-extended/xz/xz_5.4.3.bb:do_populate_sysroot', 'virtual:native:/home/chris/yocto-3/build/../poky/meta/recipes-extended/zip/zip_3.0.bb:do_populate_sysroot']
| NOTE: Installed into sysroot: []
| NOTE: Skipping as already exists in sysroot: ['nss-3.89', 'python-zstandard', 'wasi-sdk-native', 'clang-cross-arm', 'gtk+', 'dbus-glib', 'glibc', 'gcc-runtime', 'libgcc', 'quilt-native', 'libstd-rs', 'gtk+3', 'mesa', 'libxi', 'libxt', 'alsa-lib', 'pulseaudio', 'libevent', 'cbindgen-native', 'clang-native', 'nodejs-native', 'gnu-config-native', 'patch-native', 'pkgconfig-native', 'pseudo-native', 'python3-native', 'cargo-native', 'rust-native', 'xz-native', 'zip-native', 'pango', 'libpng', 'opkg-utils', 'wayland', 'libepoxy', 'libxfixes', 'libxkbcommon', 'fontconfig', 'libxext', 'libxrender', 'at-spi2-core', 'libxcursor', 'libxdamage', 'libxcomposite', 'gdk-pixbuf', 'libjpeg-turbo', 'cairo', 'glib-2.0', 'libxrandr', 'gobject-introspection', 'wayland-protocols', 'libx11', 'gettext-minimal-native', 'libtool-native', 'python3', 'libxcb', 'shadow-native', 'libatomic-ops', 'libxtst', 'systemd', 'shadow-sysroot', 'bluez5', 'libcap', 'libsm', 'shadow', 'bash-completion', 'ofono', 'avahi', 'dbus', 'libice', 'sbc', 'libtool', 'base-passwd', 'libsndfile1', 'base-files', 'speexdsp', 'attr-native', 'linux-libc-headers', 'rust-llvm-native', 'curl-native', 'libssh2-native', 'zlib-native', 'openssl-native', 'ca-certificates-native', 'zlib', 'libxml2-native', 'libedit-native', 'zstd-native', 'cmake-native', 'swig-native', 'libffi-native', 'binutils-native', 'util-macros', 'xorgproto', 'icu-native', 'file-native', 'c-ares-native', 'brotli-native', 'util-linux', 'libxdmcp', 'expat', 'libdrm', 'xrandr', 'vulkan-headers', 'libxxf86vm', 'libxshmfence', 'vulkan-loader', 'ncurses-native', 'libnsl2-native', 'util-linux-libuuid-native', 'bzip2-native', 'libtirpc-native', 'gdbm-native', 'sqlite3-native', 'readline-native', 'binutils-cross-arm', 'sqlite3', 'nspr', 'libxft', 'fribidi', 'freetype', 'harfbuzz', 'libxml2', 'libffi', 'xkeyboard-config', 'shared-mime-info', 'pixman', 'libpcre2', 'xtrans', 'texinfo-dummy-native', 'libxcrypt', 'bzip2', 'openssl', 'libtirpc', 'libnsl2', 'gdbm', 'readline', 'ncurses', 'util-linux-libuuid', 'xz', 'libxau', 'xcb-proto', 'libpthread-stubs', 'zstd', 'libseccomp', 'initscripts', 'kmod', 'update-rc.d', 'acl', 'libical', 'attr', 'ell', 'mobile-broadband-provider-info', 'libdaemon', 'libogg', 'flac', 'libvorbis', 'perl-native', 'libpcre2-native', 'elfutils-native', 'flex-native', 'unzip-native', 'libcap-ng', 'libpciaccess', 'icu', 'vala', 'make-native', 'libarchive-native', 'libmicrohttpd-native', 'm4-native', 'libgcrypt-native', 'gnutls-native', 'libgpg-error-native', 'libcap-native', 'gmp-native', 'libidn2-native', 'libunistring-native', 'nettle-native']
| DEBUG: Python function extend_recipe_sysroot finished
| DEBUG: Executing shell function do_configure
| WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f6d2a6ed710>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/glean-sdk/
| WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f6d2a186ed0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/glean-sdk/
| WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f6d2a187810>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/glean-sdk/
| WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f6d2a1c8110>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/glean-sdk/
| WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f6d2a1c8b50>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/glean-sdk/
| ERROR: Could not find a version that satisfies the requirement glean-sdk==53.1.0 (from versions: none)
| ERROR: No matching distribution found for glean-sdk==53.1.0
| WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fb16d0b4990>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/psutil/
| WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fb16d0ccd50>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/psutil/
| WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fb16d0cd3d0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/psutil/
| WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fb16d0cdd90>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/psutil/
| WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fb16d0ce7d0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/psutil/
| ERROR: Could not find a version that satisfies the requirement psutil<=5.9.4,>=5.4.2 (from versions: none)
| ERROR: No matching distribution found for psutil<=5.9.4,>=5.4.2
| WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fe4e9d55050>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/zstandard/
| WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fe4e9d6cd90>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/zstandard/
| WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fe4e9d6d990>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/zstandard/
| WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fe4e9d6e250>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/zstandard/
| WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fe4e9d6ec50>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/zstandard/
| ERROR: Could not find a version that satisfies the requirement zstandard<=0.19.0,>=0.11.1 (from versions: none)
| ERROR: No matching distribution found for zstandard<=0.19.0,>=0.11.1
|  0:00.22 Using Python 3.11.2 from /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/firefox-117.0/mozbuild_state/srcdirs/firefox-117.0-0208435d1ecb/_virtualenvs/build/bin/python
|  0:00.22 Adding configure options from /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/build/mozconfig
|  0:00.22   --enable-project=browser
|  0:00.22   --enable-application=browser
|  0:00.22   --enable-official-branding
|  0:00.22   --enable-release
|  0:00.22   --disable-strip
|  0:00.22   --disable-install-strip
|  0:00.22   --with-system-zlib
|  0:00.22   --with-system-libevent
|  0:00.22   --enable-system-ffi
|  0:00.22   --enable-system-pixman
|  0:00.22   --enable-mobile-optimize
|  0:00.22   --enable-pulseaudio
|  0:00.22   --disable-tests
|  0:00.22   --disable-necko-wifi
|  0:00.22   --disable-crashreporter
|  0:00.22   --disable-updater
|  0:00.22   --disable-debug-symbols
|  0:00.22   --allow-addon-sideload
|  0:00.22   --host=x86_64-linux
|  0:00.22   --target=arm-poky-linux-gnueabi
|  0:00.22   --host=x86_64-linux
|  0:00.22   --with-toolchain-prefix=arm-poky-linux-gnueabi-
|  0:00.22   --prefix=/usr
|  0:00.22   --libdir=/usr/lib
|  0:00.22   --with-system-nss
|  0:00.22   --disable-elf-hack
|  0:00.22   --enable-alsa
|  0:00.22   --enable-default-toolkit=cairo-gtk3-wayland
|  0:00.22   --enable-webrtc
|  0:00.22   --enable-optimize=-Os -fsigned-char -fno-strict-aliasing
|  0:00.22 checking for vcs source checkout... no
|  0:00.24 checking for a shell... /home/chris/yocto-3/build/tmp/hosttools/sh
|  0:00.25 checking for host system type... x86_64-pc-linux-gnu
|  0:00.25 checking for target system type... arm-poky-linux-gnueabi
|  0:00.38 checking whether cross compiling... yes
|  0:00.42 checking for Python 3... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/firefox-117.0/mozbuild_state/srcdirs/firefox-117.0-0208435d1ecb/_virtualenvs/build/bin/python (3.11.2)
|  0:00.43 checking for wget... /home/chris/yocto-3/build/tmp/hosttools/wget
|  0:00.43 checking for ccache... not found
|  0:00.43 checking for the target C compiler... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/wrapper/target-rust-cc
|  0:00.50 checking whether the target C compiler can be used... yes
|  0:00.50 checking the target C compiler version... 15.0.7
|  0:00.52 checking the target C compiler works... yes
|  0:00.52 WARNING: The value of LD is not used by this build system.
|  0:00.52 checking for the target C++ compiler... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/wrapper/target-rust-cxx
|  0:00.59 checking whether the target C++ compiler can be used... yes
|  0:00.59 checking the target C++ compiler version... 15.0.7
|  0:00.62 checking the target C++ compiler works... yes
|  0:00.62 checking for the host C compiler... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/wrapper/build-rust-cc
|  0:00.66 checking whether the host C compiler can be used... yes
|  0:00.66 checking the host C compiler version... 9.4.0
|  0:00.68 checking the host C compiler works... yes
|  0:00.69 checking for the host C++ compiler... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/wrapper/build-rust-cxx
|  0:00.72 checking whether the host C++ compiler can be used... yes
|  0:00.72 checking the host C++ compiler version... 9.4.0
|  0:00.75 checking the host C++ compiler works... yes
|  0:00.78 checking for target linker... bfd
|  0:00.79 checking for host linker... bfd
|  0:00.81 checking ARM version support in compiler... 7
|  0:00.84 checking for ARM SIMD support in compiler... yes
|  0:00.86 checking for ARM NEON support in compiler... yes
|  0:00.89 checking for 64-bit OS... no
|  0:00.92 checking for new enough STL headers from libstdc++... yes
|  0:00.93 checking for the assembler... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/arm-poky-linux-gnueabi/arm-poky-linux-gnueabi-clang
|  0:00.95 checking for ar... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/arm-poky-linux-gnueabi/arm-poky-linux-gnueabi-llvm-ar
|  0:00.95 checking for host_ar... /home/chris/yocto-3/build/tmp/hosttools/ar
|  0:00.98 checking for nm... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/arm-poky-linux-gnueabi/arm-poky-linux-gnueabi-llvm-nm
|  0:00.98 checking for pkg_config... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/pkg-config
|  0:00.98 checking for pkg-config version... 0.29.2
|  0:00.98 checking whether pkg-config is pkgconf... no
|  0:01.03 checking for strndup... yes
|  0:01.08 checking for posix_memalign... yes
|  0:01.12 checking for memalign... yes
|  0:01.17 checking for malloc_usable_size... yes
|  0:01.20 checking for stdint.h... yes
|  0:01.23 checking for inttypes.h... yes
|  0:01.27 checking for malloc.h... yes
|  0:01.30 checking for alloca.h... yes
|  0:01.32 checking for sys/byteorder.h... no
|  0:01.35 checking for getopt.h... yes
|  0:01.38 checking for unistd.h... yes
|  0:01.41 checking for nl_types.h... yes
|  0:01.43 checking for cpuid.h... no
|  0:01.46 checking for fts.h... yes
|  0:01.49 checking for sys/statvfs.h... yes
|  0:01.52 checking for sys/statfs.h... yes
|  0:01.54 checking for sys/vfs.h... yes
|  0:01.58 checking for sys/mount.h... yes
|  0:01.60 checking for sys/quota.h... yes
|  0:01.63 checking for linux/quota.h... yes
|  0:01.66 checking for linux/if_addr.h... yes
|  0:01.70 checking for linux/rtnetlink.h... yes
|  0:01.72 checking for sys/queue.h... yes
|  0:01.75 checking for sys/types.h... yes
|  0:01.78 checking for netinet/in.h... yes
|  0:01.81 checking for byteswap.h... yes
|  0:01.83 checking for memfd_create in sys/mman.h... yes
|  0:01.86 checking for linux/perf_event.h... yes
|  0:01.89 checking for perf_event_open system call... yes
|  0:01.91 checking whether the C compiler supports -Wbitfield-enum-conversion... yes
|  0:01.94 checking whether the C++ compiler supports -Wbitfield-enum-conversion... yes
|  0:01.97 checking whether the C++ compiler supports -Wdeprecated-this-capture... yes
|  0:01.99 checking whether the C compiler supports -Wformat-type-confusion... yes
|  0:02.02 checking whether the C++ compiler supports -Wformat-type-confusion... yes
|  0:02.04 checking whether the C compiler supports -Wshadow-field-in-constructor-modified... yes
|  0:02.07 checking whether the C++ compiler supports -Wshadow-field-in-constructor-modified... yes
|  0:02.10 checking whether the C compiler supports -Wtautological-constant-in-range-compare... yes
|  0:02.12 checking whether the C++ compiler supports -Wtautological-constant-in-range-compare... yes
|  0:02.15 checking whether the C compiler supports -Wno-error=tautological-type-limit-compare... yes
|  0:02.17 checking whether the C++ compiler supports -Wno-error=tautological-type-limit-compare... yes
|  0:02.20 checking whether the C compiler supports -Wunreachable-code-return... yes
|  0:02.23 checking whether the C++ compiler supports -Wunreachable-code-return... yes
|  0:02.25 checking whether the C compiler supports -Wunused-but-set-parameter... yes
|  0:02.28 checking whether the C++ compiler supports -Wunused-but-set-parameter... yes
|  0:02.31 checking whether the C compiler supports -Wclass-varargs... yes
|  0:02.33 checking whether the C++ compiler supports -Wclass-varargs... yes
|  0:02.36 checking whether the C++ compiler supports -Wempty-init-stmt... yes
|  0:02.38 checking whether the C compiler supports -Wfloat-overflow-conversion... yes
|  0:02.41 checking whether the C++ compiler supports -Wfloat-overflow-conversion... yes
|  0:02.43 checking whether the C compiler supports -Wfloat-zero-conversion... yes
|  0:02.46 checking whether the C++ compiler supports -Wfloat-zero-conversion... yes
|  0:02.49 checking whether the C compiler supports -Wloop-analysis... yes
|  0:02.51 checking whether the C++ compiler supports -Wloop-analysis... yes
|  0:02.54 checking whether the C compiler supports -Wno-range-loop-analysis... yes
|  0:02.57 checking whether the C++ compiler supports -Wno-range-loop-analysis... yes
|  0:02.59 checking whether the C++ compiler supports -Wc++2a-compat... yes
|  0:02.61 checking whether the C++ compiler supports -Wcomma-subscript... no
|  0:02.64 checking whether the C compiler supports -Wenum-compare-conditional... yes
|  0:02.66 checking whether the C++ compiler supports -Wenum-compare-conditional... yes
|  0:02.69 checking whether the C compiler supports -Wenum-float-conversion... yes
|  0:02.72 checking whether the C++ compiler supports -Wenum-float-conversion... yes
|  0:02.74 checking whether the C++ compiler supports -Wvolatile... no
|  0:02.77 checking whether the C++ compiler supports -Wno-error=deprecated... yes
|  0:02.79 checking whether the C++ compiler supports -Wno-error=deprecated-anon-enum-enum-conversion... yes
|  0:02.82 checking whether the C++ compiler supports -Wno-error=deprecated-enum-enum-conversion... yes
|  0:02.85 checking whether the C++ compiler supports -Wno-error=deprecated-pragma... yes
|  0:02.87 checking whether the C++ compiler supports -Wno-error=deprecated-this-capture... yes
|  0:02.90 checking whether the C++ compiler supports -Wcomma... yes
|  0:02.92 checking whether the C compiler supports -Wduplicated-cond... no
|  0:02.94 checking whether the C++ compiler supports -Wduplicated-cond... no
|  0:02.97 checking whether the C++ compiler supports -Wimplicit-fallthrough... yes
|  0:02.99 checking whether the C compiler supports -Wlogical-op... no
|  0:03.01 checking whether the C++ compiler supports -Wlogical-op... no
|  0:03.04 checking whether the C compiler supports -Wstring-conversion... yes
|  0:03.07 checking whether the C++ compiler supports -Wstring-conversion... yes
|  0:03.09 checking whether the C++ compiler supports -Wno-inline-new-delete... yes
|  0:03.11 checking whether the C compiler supports -Wno-error=maybe-uninitialized... no
|  0:03.13 checking whether the C++ compiler supports -Wno-error=maybe-uninitialized... no
|  0:03.16 checking whether the C compiler supports -Wno-error=deprecated-declarations... yes
|  0:03.19 checking whether the C++ compiler supports -Wno-error=deprecated-declarations... yes
|  0:03.21 checking whether the C compiler supports -Wno-error=array-bounds... yes
|  0:03.24 checking whether the C++ compiler supports -Wno-error=array-bounds... yes
|  0:03.26 checking whether the C compiler supports -Wno-error=free-nonheap-object... yes
|  0:03.29 checking whether the C++ compiler supports -Wno-error=free-nonheap-object... yes
|  0:03.31 checking whether the C compiler supports -Wno-multistatement-macros... no
|  0:03.33 checking whether the C++ compiler supports -Wno-multistatement-macros... no
|  0:03.35 checking whether the C compiler supports -Wno-error=class-memaccess... no
|  0:03.38 checking whether the C++ compiler supports -Wno-error=class-memaccess... no
|  0:03.40 checking whether the C compiler supports -Wno-error=atomic-alignment... yes
|  0:03.43 checking whether the C++ compiler supports -Wno-error=atomic-alignment... yes
|  0:03.45 checking whether the C compiler supports -Wno-error=deprecated-builtins... yes
|  0:03.48 checking whether the C++ compiler supports -Wno-error=deprecated-builtins... yes
|  0:03.50 checking whether the C compiler supports -Wformat... yes
|  0:03.53 checking whether the C++ compiler supports -Wformat... yes
|  0:03.56 checking whether the C compiler supports -Wformat-security... yes
|  0:03.58 checking whether the C++ compiler supports -Wformat-security... yes
|  0:03.60 checking whether the C compiler supports -Wformat-overflow=2... no
|  0:03.62 checking whether the C++ compiler supports -Wformat-overflow=2... no
|  0:03.65 checking whether the C compiler supports -Werror=implicit-function-declaration... yes
|  0:03.68 checking whether the C compiler supports -Wno-psabi... yes
|  0:03.71 checking whether the C++ compiler supports -Wno-psabi... yes
|  0:03.73 checking whether the C compiler supports -Wthread-safety... yes
|  0:03.76 checking whether the C++ compiler supports -Wthread-safety... yes
|  0:03.79 checking whether the C compiler supports -Wno-error=builtin-macro-redefined... yes
|  0:03.81 checking whether the C++ compiler supports -Wno-error=builtin-macro-redefined... yes
|  0:03.84 checking whether the C compiler supports -Wno-unknown-warning-option... yes
|  0:03.86 checking whether the C++ compiler supports -Wno-unknown-warning-option... yes
|  0:03.89 checking whether the C++ compiler supports -fno-sized-deallocation... yes
|  0:03.92 checking whether the C++ compiler supports -fno-aligned-new... yes
|  0:03.92 checking for llvm_profdata... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/llvm-profdata
|  0:03.94 checking for llvm-objdump... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/arm-poky-linux-gnueabi/arm-poky-linux-gnueabi-llvm-objdump
|  0:03.97 checking for readelf... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/arm-poky-linux-gnueabi/arm-poky-linux-gnueabi-llvm-readelf
|  0:04.00 checking for objcopy... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/arm-poky-linux-gnueabi/arm-poky-linux-gnueabi-llvm-objcopy
|  0:04.01 checking for alsa... yes
|  0:04.01 checking MOZ_ALSA_CFLAGS...
|  0:04.01 checking MOZ_ALSA_LIBS... -lasound
|  0:04.01 checking for libpulse... yes
|  0:04.01 checking MOZ_PULSEAUDIO_CFLAGS... -D_REENTRANT
|  0:04.01 checking MOZ_PULSEAUDIO_LIBS... -lpulse -pthread
|  0:04.01 checking for nspr >= 4.32... yes
|  0:04.01 checking NSPR_CFLAGS... -I/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot/usr/include/nspr
|  0:04.01 checking NSPR_LIBS... -lplds4 -lplc4 -lnspr4 -lpthread -ldl
|  0:04.01 checking for rustc... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/rustc
|  0:04.01 checking for cargo... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/cargo
|  0:04.11 checking rustc version... 1.68.2
|  0:04.12 checking cargo version... 1.68.1
|  0:04.37 checking for rust host triplet... x86_64-unknown-linux-gnu
|  0:04.43 checking for rust target triplet...
|  0:04.43 DEBUG: <truncated - see config.log for full output>
|  0:04.43 DEBUG: Executing: `/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/rustc --crate-type staticlib --target=thumbv7neon-unknown-linux-gnueabihf -o /tmp/conftestn81fgn10.rlib /tmp/conftesty0__eom_.rs -L /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot/usr/lib/rustlib/armv7-poky-linux-gnueabihf/lib --remap-path-prefix=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0=/usr/src/debug/firefox/117.0-r0 -Cpanic=abort -Clinker=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/wrapper/target-rust-ccld --sysroot=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot`
|  0:04.43 DEBUG: The command returned non-zero exit status 1.
|  0:04.43 DEBUG: Its error output was:
|  0:04.43 DEBUG: | error[E0461]: couldn't find crate `std` with expected target triple thumbv7neon-unknown-linux-gnueabihf
|  0:04.43 DEBUG: |   |
|  0:04.43 DEBUG: |   = note: the following crate versions were found:
|  0:04.43 DEBUG: |           crate `std`, target triple armv7-poky-linux-gnueabihf: /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot/usr/lib/rustlib/armv7-poky-linux-gnueabihf/lib/libstd.rlib
|  0:04.43 DEBUG: |           crate `std`, target triple armv7-poky-linux-gnueabihf: /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot/usr/lib/rustlib/armv7-poky-linux-gnueabihf/lib/libstd.so
|  0:04.43 DEBUG: |
|  0:04.43 DEBUG: | error: cannot find macro `println` in this scope
|  0:04.43 DEBUG: |  --> /tmp/conftesty0__eom_.rs:1:25
|  0:04.43 DEBUG: |   |
|  0:04.43 DEBUG: | 1 | pub extern fn hello() { println!("Hello world"); }
|  0:04.43 DEBUG: |   |                         ^^^^^^^
|  0:04.43 DEBUG: |
|  0:04.43 DEBUG: | error: requires `sized` lang_item
|  0:04.43 DEBUG: |
|  0:04.43 DEBUG: | error: aborting due to 3 previous errors
|  0:04.43 DEBUG: |
|  0:04.43 DEBUG: | For more information about this error, try `rustc --explain E0461`.
|  0:04.43 ERROR: Cannot compile for arm-poky-linux-gnueabi with /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/rustc
|  0:04.43 The target may be unsupported, or you may not have
|  0:04.43 a rust std library for that target installed. Try:
|  0:04.43
|  0:04.43   rustup target add thumbv7neon-unknown-linux-gnueabihf
|  0:04.43
| Creating global state directory from environment variable: /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/firefox-117.0/mozbuild_state
| Creating local state directory: /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/firefox-117.0/mozbuild_state/srcdirs/firefox-117.0-0208435d1ecb
| Could not install glean-sdk, so telemetry will not be collected. Continuing.
| Could not install psutil, so telemetry will be missing some data. Continuing.
| Could not install zstandard, so zstd archives will not be possible to extract. Continuing.
| Site not up-to-date reason: "/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/firefox-117.0/mozbuild_state/srcdirs/firefox-117.0-0208435d1ecb/_virtualenvs/build" does not exist
| *** Fix above errors and then restart with "./mach build"
| WARNING: exit code 1 from a shell command.
ERROR: Task (/home/chris/yocto-3/build/../meta-browser/meta-firefox/recipes-browser/firefox/firefox_117.0.bb:do_configure) failed with exit code '1'
NOTE: Tasks Summary: Attempted 6599 tasks of which 6582 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /home/chris/yocto-3/build/../meta-browser/meta-firefox/recipes-browser/firefox/firefox_117.0.bb:do_configure
Summary: There was 1 WARNING message.
Summary: There was 1 ERROR message, returning a non-zero exit code.
2023-09-12 12:03:33 - ERROR    - Command "/home/chris/yocto-3/poky/bitbake/bin/bitbake -c build core-image-initramfs sunxi-media-renew-vista update-image renew-image" failed with error 1
```

@OldManYellsAtCloud - I am not using meta-rust. The mickledore build just failed again similarly for me. What am I missing? 

```
|  0:13.54 checking whether the C compiler supports -Wno-error=class-memaccess... no
|  0:13.63 checking whether the C++ compiler supports -Wno-error=class-memaccess... no
|  0:13.73 checking whether the C compiler supports -Wno-error=atomic-alignment... yes
|  0:13.83 checking whether the C++ compiler supports -Wno-error=atomic-alignment... yes
|  0:13.94 checking whether the C compiler supports -Wno-error=deprecated-builtins... yes
|  0:14.04 checking whether the C++ compiler supports -Wno-error=deprecated-builtins... yes
|  0:14.14 checking whether the C compiler supports -Wformat... yes
|  0:14.25 checking whether the C++ compiler supports -Wformat... yes
|  0:14.36 checking whether the C compiler supports -Wformat-security... yes
|  0:14.46 checking whether the C++ compiler supports -Wformat-security... yes
|  0:14.55 checking whether the C compiler supports -Wformat-overflow=2... no
|  0:14.65 checking whether the C++ compiler supports -Wformat-overflow=2... no
|  0:14.79 checking whether the C compiler supports -Werror=implicit-function-declaration... yes
|  0:14.90 checking whether the C compiler supports -Wno-psabi... yes
|  0:15.00 checking whether the C++ compiler supports -Wno-psabi... yes
|  0:15.10 checking whether the C compiler supports -Wthread-safety... yes
|  0:15.21 checking whether the C++ compiler supports -Wthread-safety... yes
|  0:15.31 checking whether the C compiler supports -Wno-error=builtin-macro-redefined... yes
|  0:15.41 checking whether the C++ compiler supports -Wno-error=builtin-macro-redefined... yes
|  0:15.53 checking whether the C compiler supports -Wno-unknown-warning-option... yes
|  0:15.63 checking whether the C++ compiler supports -Wno-unknown-warning-option... yes
|  0:15.75 checking whether the C++ compiler supports -fno-sized-deallocation... yes
|  0:15.87 checking whether the C++ compiler supports -fno-aligned-new... yes
|  0:15.88 checking for llvm_profdata... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/llvm-profdata
|  0:16.01 checking for llvm-objdump... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/arm-poky-linux-gnueabi/arm-poky-linux-gnueabi-llvm-objdump
|  0:16.14 checking for readelf... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/arm-poky-linux-gnueabi/arm-poky-linux-gnueabi-llvm-readelf
|  0:16.26 checking for objcopy... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/arm-poky-linux-gnueabi/arm-poky-linux-gnueabi-llvm-objcopy
|  0:16.27 checking for alsa... yes
|  0:16.27 checking MOZ_ALSA_CFLAGS...
|  0:16.27 checking MOZ_ALSA_LIBS... -lasound
|  0:16.28 checking for libpulse... yes
|  0:16.28 checking MOZ_PULSEAUDIO_CFLAGS... -D_REENTRANT
|  0:16.29 checking MOZ_PULSEAUDIO_LIBS... -lpulse -pthread
|  0:16.29 checking for nspr >= 4.32... yes
|  0:16.29 checking NSPR_CFLAGS... -I/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot/usr/include/nspr
|  0:16.30 checking NSPR_LIBS... -lplds4 -lplc4 -lnspr4 -lpthread -ldl
|  0:16.30 checking for rustc... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/rustc
|  0:16.30 checking for cargo... /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/cargo
|  0:16.41 checking rustc version... 1.68.2
|  0:16.43 checking cargo version... 1.68.1
|  0:16.67 checking for rust host triplet... x86_64-unknown-linux-gnu
|  0:16.81 checking for rust target triplet...
|  0:16.81 DEBUG: <truncated - see config.log for full output>
|  0:16.81 DEBUG: Executing: `/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/rustc --crate-type staticlib --target=thumbv7neon-unknown-linux-gnueabihf -o /tmp/conftestxk0bz2yz.rlib /tmp/conftestp7j8fn6t.rs -L /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot/usr/lib/rustlib/armv7-poky-linux-gnueabihf/lib --remap-path-prefix=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0=/usr/src/debug/firefox/117.0-r0 -Cpanic=abort -Clinker=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/wrapper/target-rust-ccld --sysroot=/home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot`
|  0:16.81 DEBUG: The command returned non-zero exit status 1.
|  0:16.81 DEBUG: Its error output was:
|  0:16.81 DEBUG: | error[E0461]: couldn't find crate `std` with expected target triple thumbv7neon-unknown-linux-gnueabihf
|  0:16.81 DEBUG: |   |
|  0:16.81 DEBUG: |   = note: the following crate versions were found:
|  0:16.81 DEBUG: |           crate `std`, target triple armv7-poky-linux-gnueabihf: /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot/usr/lib/rustlib/armv7-poky-linux-gnueabihf/lib/libstd.rlib
|  0:16.81 DEBUG: |           crate `std`, target triple armv7-poky-linux-gnueabihf: /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot/usr/lib/rustlib/armv7-poky-linux-gnueabihf/lib/libstd.so
|  0:16.81 DEBUG: |
|  0:16.81 DEBUG: | error: cannot find macro `println` in this scope
|  0:16.81 DEBUG: |  --> /tmp/conftestp7j8fn6t.rs:1:25
|  0:16.81 DEBUG: |   |
|  0:16.81 DEBUG: | 1 | pub extern fn hello() { println!("Hello world"); }
|  0:16.81 DEBUG: |   |                         ^^^^^^^
|  0:16.81 DEBUG: |
|  0:16.81 DEBUG: | error: requires `sized` lang_item
|  0:16.81 DEBUG: |
|  0:16.81 DEBUG: | error: aborting due to 3 previous errors
|  0:16.81 DEBUG: |
|  0:16.81 DEBUG: | For more information about this error, try `rustc --explain E0461`.
|  0:16.81 ERROR: Cannot compile for arm-poky-linux-gnueabi with /home/chris/yocto-3/build/tmp/work/cortexa7t2hf-neon-poky-linux-gnueabi/firefox/117.0-r0/recipe-sysroot-native/usr/bin/rustc
|  0:16.81 The target may be unsupported, or you may not have
|  0:16.81 a rust std library for that target installed. Try:
|  0:16.81
|  0:16.81   rustup target add thumbv7neon-unknown-linux-gnueabihf
|  0:16.81
| *** Fix above errors and then restart with "./mach build"
| WARNING: exit code 1 from a shell command.
ERROR: Task (/home/chris/yocto-3/build/../meta-browser/meta-firefox/recipes-browser/firefox/firefox_117.0.bb:do_configure) failed with exit code '1
```

--- a/meta-firefox/README.md
+++ b/meta-firefox/README.md
@@ -3,16 +3,11 @@ Note from OldManYellsAtCloud
 
 This is an updated version of meta-firefox layer, with a newer version of Firefox.
 
-It is based on the meta-browser layer. It has been tested with Yocto Mickledore only, mostly due to lack of laptop power. I would expect it to work with previous version also, if meta-rust is pulled in.
-Feel free to test, and report back the results. 
-
-The layer build successfully with aarch64-poky-linux-gnu arch (tested with PinePhone). Tests with other architectures would be welcome.
+Currently supports Kirkstone, Mickledore and  Nanbield version of Yocto.
 
 Plan to upstream this in the near future. If you want to help in, feel free to solve an issue or open a new one.
 
 Make sure to add `RUST_PANIC_STRATEGY = "abort"` to local.conf before compiling Rust.
-
-In case you are building the browser for a CPU without hardware encryption support (e.g. anything before armv8, like the first gen RPi), set the following in your distro.conf: `PREFERRED_VERSION_nss = "3.89"`
 
 OpenEmbedded/Yocto BSP layer for Firefox Browser
 ================================================
@@ -32,6 +27,12 @@ This layer depends on:
   - revision: HEAD
 
 * URI: git://github.com/kraj/meta-clang
+  - branch: master (at least llvm version 16)
+  - revision: HEAD
+
+To build with Kirkstone, meta-rust layer is also required:
+
+* URI: https://github.com/meta-rust/meta-rust/
   - branch: master
   - revision: HEAD
 
@@ -41,21 +42,9 @@ Contributing
 The preferred way to contribute to this layer is to send GitHub pull requests or
 report problems in GitHub's issue tracker.
 
-Alternatively there is the classic way of review on the OpenEmbedded dev mailing
-list openembedded-devel@lists.openembedded.org (you have to be subscribed to
-post to the list). Please cc the maintainers if you send your patches.
-
 Maintainers
 -----------
-* Fabio Berton <fabio.berton@ossystems.com.br>
-* Khem Raj <raj.khem@gmail.com>
-* Otavio Salvador <otavio@ossystems.com.br>
-* Takuro Ashie <ashie@clear-code.com>
-
-When sending single patches, please use something like :
-```
-git send-email -1 -s --to openembedded-devel@lists.openembedded.org --subject-prefix='meta-browser][PATCH'
-```
+* OldManYellsAtCloud <skandigraun@gmail.com>
 
 Recipes
 -------
@@ -63,6 +52,9 @@ recipes-browser/firefox:
 Firefox browser.
 
 This recipe provides a package for the Firefox web browser.
+
+Two separate recipes are available: one for the latest stable version, and one
+for the latest ESR version.
 
 PACKAGECONFIG knobs
 -------------------

--- a/meta-firefox/classes/mozilla.bbclass
+++ b/meta-firefox/classes/mozilla.bbclass
@@ -31,11 +31,14 @@ mozilla_run_mach() {
 	export SHELL="/bin/sh"
 	export RUSTFLAGS="${RUSTFLAGS} -Cpanic=abort -Clinker=${WORKDIR}/wrapper/target-rust-ccld --sysroot=${RECIPE_SYSROOT}"
 
-	export RUST_HOST="${RUST_BUILD_SYS}"
-	export RUST_TARGET="${RUST_HOST_SYS}"
+	# For Kirkstone meta-rust layer is required, however
+	# that seems to set different target/host triplets compared
+	# to the rust used in poky (in later releases)
+	export RUST_HOST='${@bb.utils.contains("LAYERSERIES_CORENAMES", "kirkstone", "${BUILD_SYS}", "${RUST_BUILD_SYS}", d)}'
+	export RUST_TARGET='${@bb.utils.contains("LAYERSERIES_CORENAMES", "kirkstone", "${TARGET_SYS}", "${RUST_HOST_SYS}", d)}'
 
 	export BINDGEN_MFLOAT="${@bb.utils.contains('TUNE_CCARGS_MFLOAT', 'hard', '-mfloat-abi=hard', '', d)}"
-        export BINDGEN_CFLAGS="--target=${TARGET_SYS} --sysroot=${RECIPE_SYSROOT} ${BINDGEN_MFLOAT}"
+	export BINDGEN_CFLAGS="--target=${TARGET_SYS} --sysroot=${RECIPE_SYSROOT} ${BINDGEN_MFLOAT}"
 	export INSTALL_SDK=0
 	export DESTDIR="${D}"
 

--- a/meta-firefox/conf/layer.conf
+++ b/meta-firefox/conf/layer.conf
@@ -105,6 +105,6 @@ BBFILE_PATTERN_firefox-browser-layer := "^${LAYERDIR}/"
 BBFILE_PRIORITY_firefox-browser-layer = "7"
 
 LAYERVERSION_firefox-browser-layer = "2"
-LAYERSERIES_COMPAT_firefox-browser-layer = "mickledore nanbield"
+LAYERSERIES_COMPAT_firefox-browser-layer = "kirkstone mickledore nanbield"
 
 LAYERDEPENDS_firefox-browser-layer = "clang-layer core openembedded-layer"

